supplementary/alica_ros_turtlesim/alica/Expr/include/alica/DynamicPlanCreator.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <engine/IPlanCreator.h>
+
+#include <functional>
+#include <memory>
+
+namespace alica
+{
+
+class BasicBehaviour;
+
+class DynamicPlanCreator : public IPlanCreator
+{
+public:
+    DynamicPlanCreator(const std::string& defaultLibraryPath);
+    virtual ~DynamicPlanCreator();
+    std::unique_ptr<BasicPlan> createPlan(int64_t planId, PlanContext& context) override;
+
+private:
+    const std::string _libraryRelativePath{"/../../../lib/"};
```

Comment:
Although this may not be making assumptions about install directory location, it is still assuming relative location of include and lib directories, which I think is a bit dangerous.  Did you find out how ROS is figuring out the paths of plugins?